game-core/src/main/java/games/strategy/triplea/delegate/DiceRoll.java

```diff
@@ -1069,27 +1069,33 @@ private static int getSupport(
       final Predicate<UnitSupportAttachment> ruleFilter) {
     int givenSupport = 0;
     for (final List<UnitSupportAttachment> bonusType : supportsAvailable) {
+      final int bonusTypeCount = bonusType.get(0).getBonusType().getFirst();
+      int maxPerBonusType = bonusTypeCount >= 1 ? bonusTypeCount : Integer.MAX_VALUE;
       for (final UnitSupportAttachment rule : bonusType) {
         if (!ruleFilter.test(rule)) {
           continue;
         }
         final Set<UnitType> types = rule.getUnitType();
         if (types != null && types.contains(unit.getType()) && supportLeft.getInt(rule) > 0) {
-          givenSupport += rule.getBonus();
-          supportLeft.add(rule, -1);
-          final IntegerMap<Unit> supportersLeft = supportUnitsLeft.get(rule);
-          if (supportersLeft != null) {
-            final Set<Unit> supporters = supportersLeft.keySet();
-            if (!supporters.isEmpty()) {
-              final Unit u = supporters.iterator().next();
-              supportUnitsLeft.get(rule).add(u, -1);
-              if (supportUnitsLeft.get(rule).getInt(u) <= 0) {
-                supportUnitsLeft.get(rule).removeKey(u);
-              }
-              unitSupportMap.computeIfAbsent(u, i -> new IntegerMap<>()).add(unit, rule.getBonus());
+          final int numSupportToApply =
```

Comment:
Is this code covered by any existing tests? How safe it is it to update?

Reply:
Indirectly, yes. There are dice roll tests that check support being applied as well as some battle tests. I also now added a direct unit test for finding total power for support.